partner_aging/wizard/customer.py

```diff
@@ -0,0 +1,178 @@
+                     WHEN (aml.full_reconcile_id is NULL and aml.amount_residual>0) THEN aml.debit-(select coalesce(sum(apr.amount),0) from account_partial_reconcile apr where (apr.credit_move_id =aml.id or apr.debit_move_id=aml.id) and apr.create_date <= '%s')
+                     WHEN (aml.full_reconcile_id is NOT NULL) THEN aml.amount_residual END AS total,
+                ai.id as invoice_id,
+                ai.date_due as inv_date_due
+                FROM account_move_line aml
+                INNER JOIN         
+                  (     
+                   SELECT lt.id, 
+                   CASE WHEN inv.date_due is null then 0
+                   WHEN inv.id is not null THEN '%s' - inv.date_due 
+                   ELSE current_date - lt.date END AS days_due              
+                   FROM account_move_line lt LEFT JOIN account_invoice inv on lt.move_id = inv.move_id   
+                ) DaysDue       
+                ON DaysDue.id = aml.id
+                LEFT JOIN account_invoice as ai ON ai.move_id = aml.move_id
+                WHERE
+                aml.user_type_id in (select id from account_account_type where type = 'receivable')
+                and aml.date <= '%s'
+                and aml.amount_residual != 0
+                GROUP BY aml.partner_id, aml.id, ai.number, days_due, ai.user_id, ai.id
+              """ % (age_date, age_date, age_date, age_date, age_date,  # noqa
+                     age_date, age_date, age_date, age_date, age_date,
+                     age_date, age_date, age_date, age_date)
+
+        tools.drop_view_if_exists(self.env.cr, '%s' %
+                                  (self._name.replace('.', '_')))
+        self.env.cr.execute("""CREATE OR REPLACE VIEW %s AS (%s)""" %
+                            (self._name.replace('.', '_'), query))
+
+    @api.multi
+    def docopen(self):
+        """
+        @description  Open document (invoice or payment) related to the
+                      unapplied payment or outstanding balance on this line
+        """
+
+        models = self.env['ir.model.data']
+        # Get this line's invoice id
+        inv_id = self.invoice_id.id
+
+        # If this is an unapplied payment (all unapplied payments hard-coded to
+        #  -999), get the referenced voucher
+        if inv_id == -999:
+            payment = self.env['account.voucher']
+            # Get referenced customer payment
+            # (invoice_ref field is actually a payment for these)
+            voucher_id = payment.search([('number', '=', self.invoice_ref)])[0]
+            view = models.xmlid_to_object('account_voucher.view_voucher_form')
+            # Set values for form
+            view_id = view and view.id or False
+            name = 'Customer Payments'
+            res_model = 'account.voucher'
+            ctx = "{}"
```

Comment:
A string? IMO this should be a real dict & Python will take care of string translation if/when necessary.